### PR TITLE
Fixes the VV list view

### LINKED ISSUE #5347: Associative Lists are borked in VV
That's how it looks and it shouldn't look like that. Yes, I checked on different lists, I couldn't get a single one to work.

### PR DESCRIPTION
**Pull request by Crazylemon64**
I think I might've accidentally stripped this when looking at this conditional wrong, when refactoring the file a while back. Whoops!

Fixes #5347
:cl: Crazylemon
fix: Associative lists in VV now display properly again
/:cl:

### CODE REVIEW

File: code/datums/datumvars.dm [hunk 1]
```diff
@@ -386,7 +386,10 @@ body
 				html += "<ul>"
 				var/index = 1
 				for(var/entry in L)
-					html += debug_variable(index, L[index], level + 1)
+					if(istext(entry))
+						html += debug_variable(entry, L[index], level + 1)
```
Comment (monster860) [inline code/datums/datumvars.dm]: This wouldn't work. You need to do `L[entry]` not `L[index]` Also, not all associative lists use text as keys.

File: code/datums/datumvars.dm [hunk 2]
```diff
@@ -386,7 +386,10 @@ body
 				html += "<ul>"
 				var/index = 1
 				for(var/entry in L)
-					html += debug_variable(index, L[index], level + 1)
+					if(istext(entry))
```
Comment (monster860) [inline code/datums/datumvars.dm]: Text is still not the only possible key type.

Try `!isnum(entry) && !isnull(L[entry])`

Comment (Crazylemon64) [inline code/datums/datumvars.dm]: Alright, we're entering enhancement territory, not that I mind

Comment (Crazylemon64) [inline code/datums/datumvars.dm]: Wait, what if it's legit `null` there, like `organs_by_name` when the limb is missing?

Comment (monster860) [inline code/datums/datumvars.dm]: Then loop through the list beforehand and check if it's associative or not?

Comment (Crazylemon64) [inline code/datums/datumvars.dm]: I guess that could work, I'll give that a shot

File: code/datums/datumvars.dm [hunk 3]
```diff
@@ -385,8 +385,13 @@ body
 			else
 				html += "<ul>"
 				var/index = 1
+				var/associative = FALSE
 				for(var/entry in L)
-					html += debug_variable(index, L[index], level + 1)
+					if(!(isnum(entry) || (isnull(L[entry]) && !associative)))
+					 	associative = TRUE
```
Comment (monster860) [inline code/datums/datumvars.dm]: You have a rogue space here.

**Pull request merged**